### PR TITLE
Fix compressed write

### PR DESCRIPTION
**Pull request by rschwanhold**
### Description:
- Downsampling of compressed datasets failed. The cause was a bug in the the way compressed data was written
- `_handle_compressed_write` passed an absolute offset to `read`, which expected a relative offset. This is now fixed.

### Todos:
Make sure to delete unnecessary points or to check all before merging:
 - [x] Updated Changelog

### CODE REVIEW

File: tests/test_dataset.py [hunk 1]
```diff
@@ -2050,11 +2050,16 @@ def test_compression() -> None:
 
     # writing unaligned data to an uncompressed dataset
     write_data = (np.random.rand(3, 10, 20, 30) * 255).astype(np.uint8)
-    mag1.write(write_data)
+
+    # Writing compressed data directly to "mag1" also works, but using a View here covers an additional edge case
```
Comment (philippotto) [inline tests/test_dataset.py:2054]: Does this code "write compressed data directly"? Since `compress()` happens after the `write`, I would have thought that this is not the case.. If I'm wrong, please correct me, but otherwise, I would prefer if the test case was extended so that the compressed-write is also covered.

**Review by philippotto — CHANGES_REQUESTED** [review]
Awesome that you found this bug :+1: I'm a bit confused about the test case. See my one comment. Other than that, please update the changelog, too :rocket:

File: tests/test_dataset.py (hunk 1 above)
Comment (rschwanhold) [reply tests/test_dataset.py]: Good catch, I adjusted the wrong test cases. 

I updated the `Changelog` in a way that we can create a new release `0.6.7` once this is merged.

File: tests/test_dataset.py [hunk 2]
```diff
@@ -1500,24 +1500,25 @@ def test_writing_subset_of_compressed_data_multi_channel() -> None:
...
-    compressed_mag.write(
-        offset=(10, 20, 30), data=write_data2, allow_compressed_write=True
+    # Writing compressed data directly to "compressed_mag" also works, but using a View here covers an additional edge case
+    compressed_mag.get_view(offset=(50, 60, 70), is_bounded=False).write(
+        relative_offset=(10, 20, 30), data=write_data2, allow_compressed_write=True
     )
 
-    np.array_equal(
-        write_data2, compressed_mag.read(offset=(10, 20, 30), size=(10, 10, 10))
+    assert np.array_equal(
```
Comment (philippotto) [inline tests/test_dataset.py:1508]: whoops, good catch with the asserts :+1:

**Review by philippotto — APPROVED** [review]
Perfect!
